### PR TITLE
Fix test failure caused by updating springboot devfile

### LINKED ISSUE #3943: Java component build execution requires pom.xml
/kind bug

## What versions of software are you using?

**Operating System:**
All supported

**Output of `odo version`:**
master

## How did you run odo exactly?

```
$ odo project create wfrvjjcolc
 ✓  Project 'wfrvjjcolc' is ready for use
 ✓  New project created and now using project: wfrvjjcolc

$ odo create java-springboot --project wfrvjjcolc mkkivo
Validation
 ✓  Checking devfile existence [124215ns]
 ✓  Checking devfile compatibility [196698ns]
 ✓  Creating a devfile component from registry: DefaultDevfileRegistry [215428ns]
 ✓  Validating devfile component [301186ns]

Please use `odo push` command to create the component with source deployed

$ odo push --namespace wfrvjjcolc

Validation
 ✓  Validating the devfile [114069ns]

Creating Kubernetes resources for component mkkivo
 ✓  Waiting for component to start [1m]

Applying URL changes
 ✓  URL 8080-tcp: http://8080-tcp-mkkivo-wfrvjjcolc.apps.priti03.devcluster.openshift.com/ created

Syncing to component mkkivo
 ✓  Checking files for pushing [8ms]
 ✓  Syncing files to the component [3s]

Executing devfile commands for component mkkivo
 ✗  Executing build command "mvn clean -Dmaven.repo.local=/home/user/.m2/repository package -Dmaven.test.skip=true" [4s]
 ✗  Failed to start component with name mkkivo. Error: Failed to create the component: command execution failed: unable to execute the run command: unable to exec command [/bin/sh -c mvn clean -Dmaven.repo.local=/home/user/.m2/repository package -Dmaven.test.skip=true]: 
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.106 s
[INFO] Finished at: 2020-09-12T09:52:22Z
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/projects). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
: error while streaming command: command terminated with exit code 1
```
devfile.yaml
```
schemaVersion: 2.0.0
metadata:
  name: java-springboot
  version: 1.1.0
starterProjects:
  - name: springbootproject
    git:
      remotes:
        origin: "https://github.com/odo-devfiles/springboot-ex.git"
components:
  - name: tools
    container:
      image: quay.io/eclipse/che-java11-maven:nightly
      memoryLimit: 768Mi
      mountSources: true
      endpoints:
      - name: '8080-tcp'
        targetPort: 8080
      volumeMounts:
        - name: m2
          path: /home/user/.m2
  - name: m2
    volume:
      size: 3Gi
commands:
  - id: build
    exec:
      component: tools
      commandLine: "mvn clean -Dmaven.repo.local=/home/user/.m2/repository package -Dmaven.test.skip=true"
      group:
        kind: build
        isDefault: true
  - id: run
    exec:
      component: tools
      commandLine: "mvn -Dmaven.repo.local=/home/user/.m2/repository spring-boot:run"
      group:
        kind: run
        isDefault: true
  - id: debug
    exec:
      component: tools
      commandLine: "java -Xdebug -Xrunjdwp:server=y,transport=dt_socket,address=${DEBUG_PORT},suspend=n -jar target/*.jar"
      group:
        kind: debug
        isDefault: true
```
## Expected behavior

Component should get created but here it seems pom.xml file does not exists in the current path. 

## Any logs, error output, etc?

Manually executed on odo version
```
$ odo version
odo v2.0.0 (3fcd3670e)

Server: https://api.priti03.devcluster.openshift.com:6443
Kubernetes: v1.14.6+d7721aa
```
Even CI jobs are also failing on master https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_odo/3939/pull-ci-openshift-odo-master-v4.5-integration-e2e/1304656967607783424#1:build-log.txt%3A2653

### PR DESCRIPTION
**Pull request by adisky**
**What type of PR is this?**

/kind failing-test

**What does does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3943 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: tests/integration/devfile/cmd_devfile_push_test.go [hunk 1]
```diff
@@ -633,6 +633,7 @@ var _ = Describe("odo devfile push command tests", func() {
 			// 1) Push a generic Java project
 			helper.CmdShouldPass("odo", "create", "java-springboot", "--project", namespace, cmpName)
 			helper.CopyExample(filepath.Join("source", "devfiles", "springboot", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "springboot", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
```
Comment (kadel) [inline tests/integration/devfile/cmd_devfile_push_test.go:636]: Why we can't use devfile coming from odo devfile registry?

Comment (adisky) [reply tests/integration/devfile/cmd_devfile_push_test.go:636]: the devfile is recently updated, in this test we are checking for this particular string `helper.WaitForOutputToContain("Non-readable POM", 180, 10, session)`
which is not coming in the new devfile.

**Comment by adisky** [comment]
/retest

```
info: Loading sha256:6aa87a65480c15c2a4269e9921ea68b70f66129671a71bb2754ca25dabb8285f special-resource-operator
info: Loading sha256:40a568dccce50dff2b69dedd1c531e4591e583d71e84799eb09e831da9da6ffb cluster-etcd-operator
info: Loading sha256:5d92764de7ae29483613bec36e7041ba44465875c03738c1757c9ec759a1ce85 service-ca-operator
error: unable to read image registry.svc.ci.openshift.org/ci-op-7625j2n4/stable@sha256:c73423bf932b15b6151bf79048915be65009d7b6ac0df0a6be576169d40a1fd1: Get https://registry.svc.ci.openshift.org/v2/ci-op-7625j2n4/stable/manifests/sha256:c73423bf932b15b6151bf79048915be65009d7b6ac0df0a6be576169d40a1fd1: EOF
```

File: tests/integration/devfile/cmd_devfile_push_test.go (hunk 1 above)
Comment (adisky) [reply tests/integration/devfile/cmd_devfile_push_test.go:636]: this command `/artifacts/bin/build-container-full.sh` in the old springboot devfile uses `mvn -Dmaven.repo.local=/data/cache/.m2/repository -f ./pom.xml package -Dmaven.test.skip=true` which was explicitly specifying `pom.xml`, so this is generating the string `Non-readable POM` which is not generated by new devfile. 

IMO the check in the test needs to be updated, but for quick fix we can use the odo repo devfile.
Test i would update with this PR https://github.com/openshift/odo/pull/3799

Comment (kadel) [reply tests/integration/devfile/cmd_devfile_push_test.go:636]: ok

**Review by kadel — APPROVED** [review]

**Pull request merged**